lib/common/watchdog.c

```diff
@@ -90,21 +94,17 @@ panic_local(void)
 
     /* We're either pacemakerd, or a pacemaker daemon running as root */
 
-    if (pcmk__str_eq(panic_action, "crash", pcmk__str_casei)) {
-        sysrq_trigger('c');
-
-    } else if (pcmk__str_eq(panic_action, "sync-crash", pcmk__str_casei)) {
+    if (pcmk__starts_with(panic_action, "sync-")) {
         sync();
-        sysrq_trigger('c');
+        panic_action += strlen("sync-");
+    };
 
-    } else {
-        if (pcmk__str_eq(panic_action, "sync-reboot", pcmk__str_casei)) {
-            sync();
-        }
-        sysrq_trigger('b');
+    if (pcmk__str_eq(panic_action, "crash", pcmk__str_casei)) {
+        sysrq = 'c';
     }
-    /* reboot(RB_HALT_SYSTEM); rc = errno; */
-    reboot(RB_AUTOBOOT);
+
+    sysrq_trigger(sysrq);
+    reboot(reboot_cmd);
     rc = errno;
 
```

Comment:
Behavior is slightly changed. Probably not relevant thought.
If the env-variable is set to some bogus value starting with sync- it will now do a sync + reboot while it previously did just a reboot. Which is cleaner in a way that what comes in via env-variable is either fully
understood or there is a fallback to some defined default (just reboot).
We could crop off known tokens and set flags or things like sysrq and if in the end something unknown is left over issue a warning (maybe critical as we might be in a situation where things aren't working as they should and everything we do to output might be hanging delaying/preventing the reboot) and execute a default.
But probably overkill anyway.

Reply:
Yeah, I'm not a fan of catch-all defaults -- I'd rather check only acceptable values and ignore and warn for anything else. But this will be backported to 2.1 and that should wait for 3.0, so I'll make a note to do that later.